go.mod

```
@@ -1,6 +1,8 @@
 module github.com/elastic/beats/v7
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.7
```

```suggestion
toolchain go1.22.6
```

If we specify this it should match the one in .go-version (or it can just be removed) https://github.com/elastic/beats/blob/f4f1e3f9a4e81f41cba55939663cddc2cae67f02/.go-version#L1. Otherwise, CI is going to switch us to a Go 1.22.6 which will then download Go 1.22.7 on every build agent over and over again.

We should probably obsolete use of .go-version and gvm in favour of the toolchain directive but we haven't done that yet.

---

thanks for the detail here - i actually couldn't get this to build without having it here. i got errors about needing to commit the results of `go mod tidy`.

are you happy to approve once this is done? trying to avoid any more unnecessary long waits for CI :)